### ISSUE TITLE
External links in notebook collections go to nbviewer URLs, don't get re-directed

### ISSUE THREAD

**Issue by rgbkrk** [jupyter/nbviewer#242]
Example notebook:

http://nbviewer.ipython.org/github/ipython/ipython/blob/master/examples/Embedding/Index.ipynb

Click [embed_class_long.py](http://nbviewer.ipython.org/github/ipython/ipython/blob/master/examples/Embedding/embed_class_long.py), it links to http://nbviewer.ipython.org/github/ipython/ipython/blob/master/examples/Embedding/embed_class_long.py

rather than what the tree view will link you to:

http://nbviewer.ipython.org/github/ipython/ipython/tree/master/examples/Embedding/

While we could fix the notebook to provide the right links, it's clear the re-direct isn't working properly.

**Comment by krinsman** [comment]
I agree, this is confusing, since if one clicks the analogous link from tree view, the user gets redirected to the file's page on GitHub, rather than NBViewer's rendering of it.

I don't think this is a bug, but rather the intended behavior, see here:

https://github.com/jupyter/nbviewer/blob/master/nbviewer/providers/github/handlers.py#L252

All of those links get handled by the `GitHubBlobHandler`, which for non-iPynb files serves the file unmodified, as the code comment explains.

So I suppose the issue is that there is an inconsistency in behavior between `GitHubTreeHandler` and `GitHubBlobHandler`, and whether the two handlers should have the same or different behavior. But as it stands currently it seems both are behaving as intended.
